src/state.py

```diff
@@ -153,11 +156,33 @@ def _fetch_agent_password(model: ops.Model, config: WazuhConfig) -> str | None:
     return agent_password_content
 
 
+def _fetch_cluster_key(model: ops.Model) -> str:
+    """Fetch the Wazuh cluster key.
+
+    Args:
+        model: the Juju model.
+
+    Returns: the key for the cluster, if any.
+
+    Raises:
+        InvalidStateError: if the secret when the key should reside is invalid.
+    """
+    try:
+        cluster_key_secret = model.get_secret(label=WAZUH_CLUSTER_KEY_SECRET_LABEL)
+    except ops.SecretNotFoundError as exc:
+        raise InvalidStateError("Cluster key secret.") from exc
```

Comment:
Perhaps add `missing` at the end of the error message ?

Reply:
I'm also wondering if we shouldn't create it if it doesn't exists. The key is supposed to be here because created on install, but if it's missing what could the user do ?
On our side, we can re-create the key, and store it in the peer relation

Reply:
Recreating means a lot of additional logic when compared to relying on the install hook, that's why I decided to follow this approach. If anything were to happen, a secret with the appropriate labels could be created via cmd line

Reply:
We are dealing with this topic [here](https://github.com/canonical/wazuh-server-operator/pull/36/#discussion_r1829365143), so I'm closing this thread